### PR TITLE
feat(spell-check): add incremental-files-only option

### PR DESCRIPTION
**Pull request by HansRobo**
## Description

Add incremental-files-only option in spell-check workflow to check all files

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: spell-check/action.yaml [hunk 1]
```diff
@@ -18,6 +18,10 @@ inputs:
     description: ""
     required: false
     default: ${{ github.token }}
+  incremental-files-only:
```
Comment (kenji-miyake) [inline spell-check/action.yaml:21]: Could you update the README as well? :pray:

Comment (HansRobo) [reply spell-check/action.yaml:21]: Oh, I forgot!
I'll do :+1:

Comment (HansRobo) [reply spell-check/action.yaml:21]: Done!

File: spell-check/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## Description
 
-This action checks if the PR has miss spellings.  
+This action checks if the PR has miss spellings.
```
Comment (kenji-miyake) [inline spell-check/README.md:5]: Please refrain from updating unnecessary parts. :pray: 

```suggestion
This action checks if the PR has miss spellings.  
```

**Review by kenji-miyake — APPROVED** [review]
Thank you!

**Pull request merged**